src/java.base/share/classes/jdk/internal/jimage/ImageHeader.java

```diff
@@ -180,4 +185,39 @@ int getStringsOffset() {
         return getLocationsOffset() +
                getLocationsSize();
     }
+
+    /**
+     * Tries to determine the {@link ByteOrder} of a image file present at {@code imageFile}.
+     * The implementation reads the first few relevant bytes of the image file and compares it to
+     * the expected magic bytes of a jimage. If those bytes correspond to the magic bytes, then
+     * this method returns the ByteOrder of the image file, else it returns null.
+     *
+     * @param imageFile The path to the image file
+     * @return The ByteOrder if one could be determined. Else null.
+     * @throws IOException If any error occurred reading the contents of the image file.
+     */
+    static ByteOrder tryDetectByteOrder(final Path imageFile) throws IOException {
```

Comment:
Style-wise I think I would try to keep it consistent with the existing code if you can, meaning the noise/over-use of finals can mostly be removed if you can.

Reply:
Done. I've updated the PR to remove all the `final` usage from this part of the code and made it consistent with the current code.